src/python/pants/backend/native/subsystems/libc_dev.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.native.config.environment import HostLibcDev
+from pants.backend.native.subsystems.utils.parse_search_dirs import ParseSearchDirs
+from pants.base.hash_utils import hash_file
+from pants.option.custom_types import dir_option
+from pants.subsystem.subsystem import Subsystem
+from pants.util.memo import memoized_property
+
+
+class LibcDev(Subsystem):
+  """Subsystem to detect and provide the host's installed version of a libc "dev" package.
+
+  This subsystem exists to give a useful error message if the package isn't
+  installed, and to allow a nonstandard install location.
+  """
+
+  options_scope = 'libc'
+
+  class HostLibcDevResolutionError(Exception): pass
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(LibcDev, cls).subsystem_dependencies() + (ParseSearchDirs.scoped(cls),)
+
+  @memoized_property
+  def _parse_search_dirs(self):
+    return ParseSearchDirs.scoped_instance(self)
+
+  @classmethod
+  def register_options(cls, register):
+    super(LibcDev, cls).register_options(register)
+
+    register('--libc-dir', type=dir_option, default='/usr/lib', advanced=True,
```

Comment:
Should both of these options be fingerprinted?

Reply:
Yes!! Will fix and keep that in mind in the future.

Reply:
Done in 5a1665b!